src/Microsoft.TemplateEngine.TemplateLocalizer/Commands/Export/ExportCommand.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                Name = "--DryRun",
+                Description = LocalizableStrings.command_export_help_dryrun_description,
+            });
+            exportCommand.Handler = this;
+
+            return exportCommand;
+        }
+
+        protected override async Task<int> Execute(ExportCommandArgs args, CancellationToken cancellationToken)
+        {
+            // Upgrade the cancellation token to allow also cancelling within this method.
+            using CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            cancellationToken = cts.Token;
+
+            bool failed = false;
+            List<FileInfo> templateJsonFiles = new List<FileInfo>();
+
+            foreach (TemplateJsonProvider templateJsonProvider in args.TemplateJsonProviders)
+            {
+                int filesBeforeAdd = templateJsonFiles.Count;
+                templateJsonFiles.AddRange(templateJsonProvider.GetTemplateJsonFiles(args.SearchSubdirectories));
+
+                if (filesBeforeAdd == templateJsonFiles.Count)
+                {
+                    // No new files has been added by this provider. This is an indication of a bad input.
+                    Logger.LogError(LocalizableStrings.command_export_log_templateJsonNotFound, templateJsonProvider.Path);
+                    failed = true;
+                }
+            }
+
+            if (failed)
+            {
+                Logger.LogError(LocalizableStrings.generic_log_commandExecutionFailed, CommandName);
+                return 1;
+            }
+
+            ExportOptions exportOptions = new ExportOptions()
+            {
+                Languages = args.Languages,
+                DryRun = args.DryRun,
+            };
+
+            List<ExportResult> exportResults = new List<ExportResult>();
+            List<Task<ExportResult>> runningExportTasks = new List<Task<ExportResult>>(templateJsonFiles.Count);
+            foreach (FileInfo templateJsonFile in templateJsonFiles)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    break;
+                }
+
+                string templateDirectory = Path.GetDirectoryName(templateJsonFile.FullName) ?? string.Empty;
+                exportOptions.TargetDirectory = Path.Combine(templateDirectory, "localize");
+
+                runningExportTasks.Add(new Core.TemplateLocalizer(Logger).ExportLocalizationFiles(templateJsonFile.FullName, exportOptions, cancellationToken));
+
+                if (runningExportTasks.Count == ConcurrencyLevel)
+                {
+                    // We have reached the concurrency limit. Wait for one of the tasks to finish before issuing new ones.
+                    Task<ExportResult>? completedTask = null;
+
+                    try
+                    {
+                        completedTask = await Task.WhenAny(runningExportTasks).ConfigureAwait(false);
```

Comment:
i'm probably missing something, at a glance this block looks like...
`exportResults.Add(await Task.WhenAny(runningExportTasks).ConfigureAwait(false));`

Reply:
We need a reference to the completed task to be able to remove it from the list `runningExportTasks`.